src/VisualStudio/Core/Def/LanguageService/AbstractPackage`2.cs

```diff
@@ -64,11 +55,28 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             return _languageService.ComAggregate;
         });
 
-        await shell.LoadPackageAsync(Guids.RoslynPackageId);
+        var shell = await GetServiceAsync<SVsShell, IVsShell7>(throwOnFailure: true, cancellationToken).ConfigureAwait(true);
+
+        await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+        foreach (var editorFactory in CreateEditorFactories())
+        {
+            RegisterEditorFactory(editorFactory);
+        }
 
         var miscellaneousFilesWorkspace = this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
```

Comment:
This faults in a bunch of dlls and would be better on background thread.

Reply:
This (and the one in RoslynPackage) need to be done during InitializeAsync, but I'll look to see if it can be done berfore transitioning to the main thread

Reply:
moved to the bg thread